build/mac/pkg-scripts/postinstall

```diff
@@ -6,11 +6,26 @@
 installerPath=$1
 installationPath=$2
 
-# TODO: ugly to assume the name of the app, especially with multi-channel.
-# Luckily for now, release channel is the only channel with a pkg installer.
-installationAppPath="$installationPath/Brave Browser.app"
+# Detect if installer contained a referral promotion code within the filename
+# Will also get a channel-safe version of the application name (based on the .pkg file name)
 installerPathPromoCodeRegex='.+-(([a-zA-Z0-9]{3}[0-9]{3})|([a-zA-Z]{1,}-[a-zA-Z]{1,}))([[:blank:]]?\([0-9]+\))?\.pkg$'
-userDataDir="$HOME/Library/Application Support/BraveSoftware/Brave-Browser"
+if [[ $installerPath =~ $installerPathPromoCodeRegex ]]; then
+  echo "Installer path matched for promo code"
+  n=${#BASH_REMATCH[*]}
+  if [ $n -ge 1 ]; then
+    promoCode=${BASH_REMATCH[1]}
+    echo "Got promo code: $promoCode"
+  fi
+  appName=$(basename "$1" | sed "s/.pkg//" | sed "s/ /\-/g" | sed "s/-${promoCode}//")
+else
+  echo "Installer path did not match for promo code"
+  promoCode=""
+  appName=$(basename "$1" | sed "s/.pkg//" | sed "s/ /\-/g")
+fi
+
+appNameNoDash=$(echo $appName | sed "s/-/\ /g")
```

Comment:
It seems like sometimes we're escaping the minus sign and space for sed replacement and sometimes we're not. Wasn't sure if that was intentional or not? I didn't think you needed to escape for space at least, but maybe I'm wrong.

Reply:
oh yeah - good catch. There were various states of this script where the `sed` input had quotes and didn't have quotes. This is an artifact of that. Let me remove the extra escape after testing it and then amend what we have here 😄